File: PHP/LonelyInteger.php
<?php
/**
 * @version PHP 7.3.29
 * @author MARIO SAM <eu@mariosam.com.br>
 * @see I would love to work with you instead solving web code tests: hire me!
 */
namespace PHP;

class LonelyInteger {

    /*
     * Complete the 'lonelyinteger' function below.
     *
     * The function is expected to return an INTEGER.
     * The function accepts INTEGER_ARRAY a as parameter.
     */
    function lonelyinteger($a) {
        //coloca os dados em ordem crescente
        sort($a);

        //inicializa resultado
        $res = $a[0];

        //verifica se o num atual eh diferente do anterior e do proximo
        for ( $i = 1; $i < count($a); $i++ ) {
            if ( $i == count($a)-1 || $a[$i] != $a[$i-1] && $a[$i] != $a[$i+1] ) {
                $res = $a[$i];
                break;
            }
        }

        return $res;
    }

    /*
    $fptr = fopen(getenv("OUTPUT_PATH"), "w");
    $n = intval(trim(fgets(STDIN)));
    $a_temp = rtrim(fgets(STDIN));
    $a = array_map('intval', preg_split('/ /', $a_temp, -1, PREG_SPLIT_NO_EMPTY));

    $result = lonelyinteger($a);
    fwrite($fptr, $result . "\n");
    fclose($fptr);
    */
}

File: JAVA/LonelyInteger.java
/**
 * @version JAVA
 * @author MARIO SAM <eu@mariosam.com.br>
 * @see I would love to work with you instead solving web code tests: hire me!
 */
package JAVA;

import java.io.BufferedReader;
import java.io.BufferedWriter;
import java.io.FileWriter;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.Collections;
import java.util.List;
import java.util.stream.Stream;
import static java.util.stream.Collectors.toList;

public class LonelyInteger {
    
    public static void main(String[] args) throws IOException {
        BufferedReader bufferedReader = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter bufferedWriter = new BufferedWriter(new FileWriter(System.getenv("OUTPUT_PATH")));

        //int n = Integer.parseInt(bufferedReader.readLine().trim());

        List<Integer> a = Stream.of(bufferedReader.readLine().replaceAll("\\s+$", "").split(" "))
            .map(Integer::parseInt)
            .collect(toList());

        int result = lonelyinteger(a);

        bufferedWriter.write(String.valueOf(result));
        bufferedWriter.newLine();

        bufferedReader.close();
        bufferedWriter.close();
    }

    /*
     * Complete the 'lonelyinteger' function below.
     *
     * The function is expected to return an INTEGER.
     * The function accepts INTEGER_ARRAY a as parameter.
     */
    public static int lonelyinteger(List<Integer> a) {
        //coloca os dados em ordem crescente
        Collections.sort( a );

        //inicializa resultado
        int res = a.get(0);

        //verifica se o num atual eh diferente do anterior e do proximo
        for ( int i = 1; i < a.size(); i++ ) {
            if ( i == a.size()-1 || a.get(i) != a.get(i-1) && a.get(i) != a.get(i+1) ) {
                res = a.get(i);
                break;
            }
        }

        return res;
    }
}
